tools/msconvert/msconvert_raw.xml

```diff
@@ -1,396 +1,282 @@
-      <when value="full">
-        <param type="select" name="binary_encoding" label="Binary Encoding Precision">
-          <option value="64" selected="true">64</option>
-          <option value="32">32</option>
-        </param>
-        <param type="select" name="mz_encoding" label="m/z Encoding Precision">
-          <option value="64" selected="true">64</option>
-          <option value="32">32</option>
-        </param>
-        <param type="select" name="intensity_encoding" label="Intensity Encoding Precision" value="32">
-          <option value="64">64</option>
-          <option value="32" selected="true">32</option>
-        </param>
-        <param type="boolean" name="zlib" label="Use zlib">
-        </param>
-      </when>
-    </conditional>
-
+    <!-- TODO: add Bruker, Waters, Agilent types -->
+    <param format="mzml,mzxml,mz5,mgf,wiff,raw" name="input" type="data" label="Input unrefined MS data" />
 
+    <expand macro="msconvertInputParameters" />
   </inputs>
-  <outputs>
-    <data format="mzml" name="output" label="${input.name.rsplit('.',1)[0]}.${output_type}" >
-      <change_format>
-        <when input="output_type" value="mzXML" format="mzxml" />
-        <when input="output_type" value="unindexed_mzXML" format="mzxml" />
-        <when input="output_type" value="ms2" format="ms2" />
-        <when input="output_type" value="mgf" format="mgf" />
-      </change_format>
-    </data>
-  </outputs>
+
+  <expand macro="msconvertOutput" />
+
+  <tests>
+    <test>
+      <param name="input" value="small.mzML" />
+      <param name="output_type" value="mzML" />
+      <param name="pick_peaks" value="true" />
+      <param name="pick_peaks_algorithm" value="cwt" />
+      <param name="pick_peaks_ms_levels" value="1-" />
+      <output name="output" file="small-peakpicking-cwt-allMS.mzML" />
+    </test>
+
+    <!-- this data file only has profile MS1, so the result is the same -->
+    <test>
+      <param name="input" value="small.mzML" />
+      <param name="output_type" value="mzML" />
+      <param name="pick_peaks" value="true" />
+      <param name="pick_peaks_algorithm" value="cwt" />
+      <param name="pick_peaks_ms_levels" value="1" />
+      <output name="output" file="small-peakpicking-cwt-allMS.mzML" /> 
+    </test>
+    <test>
+      <param name="input" value="small-peakpicking-cwt-allMS.mzML" />
+      <param name="output_type" value="mz5" />
+      <param name="mz_encoding" value="64" />
+      <param name="intensity_encoding" value="64" />
+      <output name="output" file="small-zlib-64.mz5" compare="sim_size" delta="100" />
+    </test>
+    <test>
+      <param name="input" value="small-peakpicking-cwt-allMS.mzML" />
+      <param name="output_type" value="mzXML" />
+      <param name="mz_encoding" value="32" />
+      <param name="intensity_encoding" value="32" />
+      <output name="output" file="small-zlib-32.mzXML" />
+    </test>
+    <!-- TODO: how to test gzipped output?
```

Comment:
Yes, I'm fine with the way it is (commented out). I'd prefer to leave the option to compress in though, in case someone just wants to use it to convert a bunch of data into a reasonably compact form. Pwiz-based tools can consume the file without decompressing first (which is what I assume the "associated converter" would do).